TenancyInformationApi/V1/Gateways/TenancyGateway.cs

```diff
@@ -114,7 +113,7 @@ where string.IsNullOrEmpty(addressQuery)
                 where string.IsNullOrEmpty(postcodeQuery)
                       || EF.Functions.ILike(property.Postcode.Replace(" ", ""), postcodeSearchPattern)
                 where string.IsNullOrEmpty(propertyReference)
-                      || EF.Functions.ILike(property.PropertyReference.Replace(" ", ""), propertyReference)
+                      || EF.Functions.ILike(property.PropertyReference, propertyReference)
```

Comment:
Looks fine, except I would say, use  "where agreementType.PropertyReference" rather than property.PropertyReference. 
I think some property references will not have tenancy agreement records